packages/framework/src/client/scheduler.js

```diff
@@ -1,109 +1,148 @@
+				this.timerId = null;
+			}
+		};
 	}
 
 	/**
 	 * Finds a request that is either scheduled or overdue.
-	 * @param {string} resourceName The name of the resource to be read
+	 * @param {string} resourceName The name of the resource for the operation
+	 * @param {string} operation The name of the operation to be performed (defaults to 'read')
+	 * @param {Date} now The current time.
 	 * @return {ResourceRequest} The scheduled request, or null if none found
 	 */
-	getScheduledRequest = ( resourceName ) => {
+	getScheduledRequest = ( resourceName, operation = DEFAULT_READ_OPERATION, now = new Date() ) => {
 		return find( this.requests, ( r ) => {
-			if ( resourceName === r.resourceName ) {
-				const status = r.getStatus();
-				return STATUS.scheduled === status || STATUS.overdue === status;
-			}
-			return false;
+			const status = r.getStatus( now );
+			return (
+				resourceName === r.resourceName &&
+				operation === r.operation &&
+				( STATUS.scheduled === status || STATUS.overdue === status )
+			);
 		} );
 	}
 
 	/**
 	 * Finds requests that are in flight by resourceName
 	 * @param {string} resourceName The name of the resource to check
+	 * @param {string} operation The name of the operation to be performed (defaults to 'read')
+	 * @param {Date} now The current time.
 	 * @return {Array} Any requests for the given resource which are currently in flight.
 	 */
-	getInFlightRequests = ( resourceName ) => {
+	getInFlightRequests = ( resourceName, operation = DEFAULT_READ_OPERATION, now = new Date() ) => {
 		return this.requests.filter( ( r ) => {
-			return ( resourceName === r.resourceName && STATUS.inFlight === r.getStatus() );
+			return (
+				resourceName === r.resourceName &&
+				operation === r.operation &&
+				STATUS.inFlight === r.getStatus( now )
+			);
 		} );
 	}
 
 	/**
-	 * Schedule a read of a resource name according to the requirement set given.
-	 * @param {string} resourceName The name of the resource to be read
+	 * Schedule a operation on a resource name according to the requirement set given.
 	 * @param {Object} requirement The set of requirements (freshness, timeout)
 	 * @param {Object} resourceState The current snapshot of resource state
+	 * @param {string} resourceName The name of the resource for the operation
+	 * @param {string} operation The name of the operation to be performed (defaults to 'read')
+	 * @param {Object} data The data to be sent for the operation (defaults to null)
 	 * @param {Date} now The current time.
 	 */
-	scheduleRequest = ( resourceName, requirement, resourceState, now = new Date() ) => {
-		if ( this.getInFlightRequests( resourceName ).length > 0 ) {
-			// Do nothing, there's already a request in flight.
+	scheduleRequest = (
+		requirement,
+		resourceState,
+		resourceName,
+		operation = DEFAULT_READ_OPERATION,
+		data = null,
+		now = new Date()
+	) => {
+		if ( this.getInFlightRequests( resourceName, operation ).reduce( ( request ) => {
```

Comment:
You can use [`lodash.find`](https://lodash.com/docs/4.17.11#find) with a custom predicate, or even the built-in [`Array.prototype.find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find). Using `reduce` here feels wrong.

Reply:
Yeah that would be better. This code kind of morphed from another implementation to this one and I didn't think of how I could simplify it. Thanks!

Reply:
Updated in fcd3cab5d80d44d6f32d2514104adad4b82508bd